intents/intent_data_transaction_contract_abi.go

```diff
@@ -0,0 +1,142 @@
+package intents
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/0xsequence/ethkit/ethcoder"
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+)
+
+type contractCallType struct {
+	Abi  string `json:"abi"`
+	Func string `json:"func"`
+	Args []any  `json:"args"`
+}
+
+func EncodeContractCall(data *contractCallType) (string, error) {
+	// Get the method from the abi
+	method, _, err := getMethodFromAbi(data.Abi, data.Func)
+	if err != nil {
+		return "", err
+	}
+
+	enc := make([]string, len(data.Args))
+
+	// String args can be used right away, but any nested
+	// `contractCallType` must be handled recursively
+	for i, arg := range data.Args {
+		switch arg := arg.(type) {
+		case string:
+			enc[i] = arg
+
+		case map[string]interface{}:
+			nst := arg
+
+			var funcName string
+			if v, ok := nst["func"].(string); ok {
+				funcName = v
+			}
+
+			args, ok := nst["args"].([]interface{})
+			if !ok {
+				return "", fmt.Errorf("nested args expected to be an array")
+			}
+
+			enc[i], err = EncodeContractCall(&contractCallType{
+				Abi:  nst["abi"].(string), // TODO: should error check this..? or will it return "" ?
```

Comment:
is this safe? or should we do check first as mentioned in TODO ..?

Reply:
[This is not safe and will panic](https://go.dev/play/p/33xqDSZbyGg). The two-return form must be used to default to `""`:

```
abi, _ := nst["abi"].(string)
```